core/src/main/java/org/apache/calcite/sql/SqlPivot.java

```diff
@@ -195,5 +197,17 @@ static class Operator extends SqlSpecialOperator {
     Operator(SqlKind kind) {
       super(kind.name(), kind);
     }
+
+    @Override public SqlCall createCall(
+        @Nullable SqlLiteral functionQualifier,
+        SqlParserPos pos,
+        @Nullable SqlNode... operands) {
+      assert operands.length == 4;
+      pos = pos.plusAll(operands);
```

Comment:
Why do you add `operands` position to `pos`? I don't see other `SqlOperator#createCall`s are doing this.

Reply:
`SqlOperator#createCall` does that. It seemed useful to pass operands for the parser position.

Reply:
I debugged your test case, and found an interesting result. The `pos` for original `SqlPivot` is from 'line 3 column 1, to line 3 column 54', which represents the whole `PIVOT (...)`. However, after copying using `createCall`, it changes to 'line 2 column 7, to line 3 column 54'. The reason for this is the first operand is the `query` which does not count into the original pos.
Hence, for this case, I think we should not add this line, since it gives the wrong result. WDYT?

Reply:
Thanks for investigating it. In this case, it makes sense to remove this line, I made this change and squashed it with the existing commit.